### PR TITLE
FISH-8359 Upgrade Concurrency TCK to 3.1

### PR DESCRIPTION
**Pull request by aubi**
Update runner for concurrency to changed TCK.

TCK moved from testng to JUnit and changed dependencies versions.

Locally the TCK doesn't finish completely correctly as there are still some unfinished features, mainly Flow support:

```
Tests run: 242, Failures: 11, Errors: 4, Skipped: 24
```

### CODE REVIEW

File: concurrent-tck/pom.xml [hunk 1]
```diff
@@ -10,13 +10,13 @@
...
-        <maven.compiler.target>11</maven.compiler.target>
+        <maven.compiler.source>21</maven.compiler.source>
+        <maven.compiler.target>21</maven.compiler.target>
         
         <!-- Arquillian Dependencies -->
-        <payara.arquillian.container.version>3.0.alpha6</payara.arquillian.container.version>
+        <payara.arquillian.container.version>3.0.alpha8</payara.arquillian.container.version>
         
-        <com.beust-jcommander.version>1.82</com.beust-jcommander.version>
+        <!-- com.beust-jcommander.version>1.82</com.beust-jcommander.version -->
```
Comment (Pandrex247) [inline concurrent-tck/pom.xml:19]: If we're not using it I'd rather just delete it
```suggestion
```

File: concurrent-tck/pom.xml [hunk 2]
```diff
@@ -39,61 +39,67 @@
...
-            <artifactId>arquillian-testng-container</artifactId>
+            <groupId>org.junit.platform</groupId>
+            <artifactId>junit-platform-launcher</artifactId>
+            <version>1.10.2</version>
             <!--            <scope>compile</scope>-->
         </dependency>
         <dependency>
+            <groupId>org.jboss.arquillian.junit5</groupId>
+            <artifactId>arquillian-junit5-container</artifactId>
+            <!--            <scope>compile</scope>-->
```
Comment (Pandrex247) [inline concurrent-tck/pom.xml:71]: Pointless comment
```suggestion
```

File: concurrent-tck/pom.xml [hunk 3]
```diff
@@ -39,61 +39,67 @@
...
+            <version>5.10.2</version>
             <!--            <scope>compile</scope>-->
         </dependency>
         <dependency>
-            <groupId>org.jboss.arquillian.testng</groupId>
-            <artifactId>arquillian-testng-container</artifactId>
+            <groupId>org.junit.platform</groupId>
+            <artifactId>junit-platform-launcher</artifactId>
+            <version>1.10.2</version>
             <!--            <scope>compile</scope>-->
```
Comment (Pandrex247) [inline concurrent-tck/pom.xml:66]: Pointless comment
```suggestion
            <version>1.10.2</version>
```

File: concurrent-tck/pom.xml [hunk 4]
```diff
@@ -39,61 +39,67 @@
...
         </dependency>
         <!-- Test frameworks -->
         <dependency>
-            <groupId>org.testng</groupId>
-            <artifactId>testng</artifactId>
-            <version>6.14.3</version>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <version>5.10.2</version>
             <!--            <scope>compile</scope>-->
```
Comment (Pandrex247) [inline concurrent-tck/pom.xml:60]: Pointless comment
```suggestion
            <version>5.10.2</version>
```

File: concurrent-tck/pom.xml [hunk 5]
```diff
@@ -39,61 +39,67 @@
...
-        </dependency>
+        </dependency -->
         
         <!-- Remote Payara server -->
         <dependency>
             <groupId>fish.payara.arquillian</groupId>
             <artifactId>arquillian-payara-server-remote</artifactId>
             <version>${payara.arquillian.container.version}</version>
-            <scope>test</scope>
+            <!-- scope>test</scope -->
```
Comment (Pandrex247) [inline concurrent-tck/pom.xml:84]: Compile scope?
```suggestion
            <scope>test</scope>
```

File: concurrent-tck/pom.xml [hunk 6]
```diff
@@ -39,61 +39,67 @@
...
         </dependency>
         <dependency>
             <groupId>fish.payara.arquillian</groupId>
             <artifactId>payara-client-ee9</artifactId>
-            <scope>test</scope>
+            <!-- scope>test</scope -->
```
Comment (Pandrex247) [inline concurrent-tck/pom.xml:89]: Compile scope?
```suggestion
            <scope>test</scope>
```

File: concurrent-tck/pom.xml [hunk 7]
```diff
@@ -148,14 +154,25 @@
...
                     <systemPropertyVariables>
                     </systemPropertyVariables>
-                    <suiteXmlFiles>
-                        <suiteXmlFile>${tck-suite.xml}</suiteXmlFile>
-                    </suiteXmlFiles>
-                    <testSourceDirectory>${basedir}${file.separarator}src${file.separarator}main${file.separarator}java${file.separarator}</testSourceDirectory>
+                    <groups>full</groups>                        <!-- Groups to include i.e. web/full -->
+                    <!-- <includes>
+                        <include>ee.jakarta.tck.concurrent.spec.ManagedThreadFactory.injected.*Tests</include>
+                    </includes> -->
```
Comment (Pandrex247) [inline concurrent-tck/pom.xml:171]: Why is this commented out? An explanation in the pom wouldn't go amiss.

**Review by Pandrex247 — CHANGES_REQUESTED** [review]

File: concurrent-tck/pom.xml (hunk 7 above)
Comment (aubi) [reply concurrent-tck/pom.xml:171]: I am adding a comment; this is useful when debugging some particular tests. I would definitely forgot, how to write it :-)

**Review by Pandrex247 — APPROVED** [review]

**Pull request merged**